### PR TITLE
Artembo/auditlog

### PR DESCRIPTION
**Pull request by artembo**
Implement an audit log to track the system state modifications

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -174,14 +186,18 @@ def put(self):
                                                    rock_content=rockspec, action='add')
 
         if patched_manifest is not None:
+            package_file_obj = BytesIO(package)
+            self.client.upload_fileobj(package_file_obj, self.bucket, f'{S3_ROCKS_FOLDER}{file_name}')
+            self.audit_log(f'put {file_name} - {message}', package_file_obj)
```
Comment (rosik) [inline app.py]: Unsuccessful attempts should be logged too

File: app.py [hunk 2]
```diff
@@ -174,14 +186,18 @@ def put(self):
...
 
-            self.client.upload_fileobj(BytesIO(package), self.bucket, f'{S3_ROCKS_FOLDER}{file_name}')
-
-            self.client.upload_fileobj(BytesIO(str.encode(patched_manifest)), self.bucket, f'{S3_ROCKS_FOLDER}manifest')
+            manifest_file_obj = BytesIO(str.encode(patched_manifest))
+            self.client.upload_fileobj(manifest_file_obj, self.bucket, f'{S3_ROCKS_FOLDER}manifest')
+            self.audit_log(f'update manifest', manifest_file_obj)
```
Comment (rosik) [inline app.py:195]: Duplicate message

File: app.py [hunk 3]
```diff
@@ -174,14 +186,18 @@ def put(self):
...
 
         return response_message(message)
 
     def get(self, path='/'):
+        self.audit_log('event')
```
Comment (rosik) [inline app.py:200]: Flood

Comment (artembo) [reply app.py]: removed

**Review by rosik — APPROVED** [review]

**Pull request merged**